### PR TITLE
feat(cdp): Use cyclotron part 2

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

The first PR refactored the code to make it easier to enqueue and read from Cyclotron instead of Kafka. This implements it

## Changes

* Adds the extra work needed to allow writing and reading from Cyclotron postgres instead of kafka
* Works e2e with flags so we can slowly start testing it

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

**Comment by bretthoerner** [comment]
I got this at runtime:

```
Error: Cannot find module './manager'
Require stack:
- /Users/brett/Development/posthog/posthog/plugin-server/node_modules/.pnpm/file+..+rust+cyclotron-node/node_modules/@posthog/cyclotron/dist/index.js
- /Users/brett/Development/posthog/posthog/plugin-server/src/cdp/cdp-consumers.ts
- /Users/brett/Development/posthog/posthog/plugin-server/src/main/pluginsServer.ts
- /Users/brett/Development/posthog/posthog/plugin-server/src/index.ts
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Function.Module._load (node:internal/modules/cjs/loader:922:27)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:121:18)
    at Object.<anonymous> (/Users/brett/Development/posthog/posthog/plugin-server/node_modules/.pnpm/file+..+rust+cyclotron-node/node_modules/@posthog/cyclotron/src/index.ts:1:1)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Object.newLoader [as .js] (/Users/brett/Development/posthog/posthog/plugin-server/node_modules/.pnpm/pirates@4.0.5/node_modules/pirates/lib/index.js:141:7)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Function.Module._load (node:internal/modules/cjs/loader:960:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/brett/Development/posthog/posthog/plugin-server/node_modules/.pnpm/file+..+rust+cyclotron-node/node_modules/@posthog/cyclotron/dist/index.js',
    '/Users/brett/Development/posthog/posthog/plugin-server/src/cdp/cdp-consumers.ts',
    '/Users/brett/Development/posthog/posthog/plugin-server/src/main/pluginsServer.ts',
    '/Users/brett/Development/posthog/posthog/plugin-server/src/index.ts'
  ]
}
```

Even after whatever `bin/start` runs and `pnpm i`, my cyclotron install was still this:

```
# plugin-server/node_modules/@posthog/cyclotron
$ find .
.
./dist
./dist/index.js
./dist/index.js.map
./dist/index.d.ts
./index.node
./package.json
```

Did you have to bust the cache locally? I did `pnpm store prune` which was probably heavy handed, and I don't recommend it. Maybe `pnpm update @posthog/cyclotron --force` alone would have worked. Either way it seems like our install process doesn't detect/install changes in `cyclotron-node`?

**Comment by bretthoerner** [comment]
Seemed to work for me otherwise, though. I have a Webhook defined, they all fired, went back to hog, and then were cleaned up by the janitor.

Metrics are happy:
```
cyclotron_janitor_completed_jobs{janitor_id="default_janitor_id",shard_id="default"} 145

#...

cyclotron_fetch_jobs_completed{shard_id="default",worker_id="default_worker_id",queue_served="fetch"} 145
```

**Review by mariusandra — CHANGES_REQUESTED** [review]
Tested locally, everything seems to work. Two issues worth addressing though:
- I think I found a bug with the cyclotron/kafka split (comment inline)
- `bin/start-cyclotron` should also use the new env.

File: plugin-server/src/cdp/cdp-consumers.ts [hunk 1]
```diff
@@ -205,20 +211,23 @@ abstract class CdpConsumerBase {
...
-        // For kafka style this is overkill to enqueue this way but it simplifies migrating to the new system
+    protected async queueInvocationToKafka(invocation: HogFunctionInvocation) {
+        // NOTE: WE keep the queueParams args as kafka land still needs them
+        const serializedInvocation: HogFunctionInvocationSerialized = {
+            ...invocation,
+            hogFunctionId: invocation.hogFunction.id,
+        }
 
-        const serializedInvocation = serializeInvocation(invocation)
+        delete (serializedInvocation as any).hogFunction
```
[the remark this answers is not in the document]
Comment (benjackwhite) [reply plugin-server/src/cdp/cdp-consumers.ts:230]: Same same. Given this is temporary I'm not going to optimise around it if thats okay

**Review by mariusandra — APPROVED** [review]
Go forth and multiply!

**Pull request merged**